static/app/views/monitors/components/overviewTimeline/utils.tsx

```diff
@@ -24,3 +26,38 @@ export function getStartFromTimeWindow(end: Date, timeWindow: TimeWindow): Date
 
   return start.toDate();
 }
+
+/**
+ * Given two env mappings e.g. {prod: {ok: 1, missed: 0, etc..}}
+ * combines job status counts
+ * @param envMappingA
+ * @param envMappingB
+ * @returns The merged environment mapping
+ */
+export function mergeEnvMappings(
+  envMappingA: MonitorBucketEnvMapping,
+  envMappingB: MonitorBucketEnvMapping
+) {
+  const combinedEnvs = new Set([
+    ...Object.keys(envMappingA),
+    ...Object.keys(envMappingB),
+  ]);
+  return [...combinedEnvs].reduce((mergedEnvs, env) => {
+    const mergedStatusCounts = {};
+    for (const status of Object.values(CheckInStatus)) {
```

Comment:
Will this produce extraneous buckets, since there are some status's in here that we don't care about in the stats?